### PR TITLE
corected method

### PR DESCRIPTION
**Pull request by layzimayner**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -1,19 +1,28 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.HashMap;
+import java.util.Map;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private Map<K, V> storage;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]: [Don't use any kind of List, Set or Map](https://github.com/mate-academy/jv-storage/blob/master/checklist.md#dont-use-any-kind-of-list-set-or-map)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,54 @@
...
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_CAPACITY = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[INITIAL_CAPACITY];
+        values = (V[]) new Object[INITIAL_CAPACITY];
+        size = 0;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:14]:
```suggestion
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,54 @@
...
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private void resize() {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:48]: remove

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,54 @@
...
+    public StorageImpl() {
+        keys = (K[]) new Object[INITIAL_CAPACITY];
+        values = (V[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == key || (keys[i] != null && keys[i].equals(key))) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:20]: duplication of code, create separate indexOf method

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,53 @@
...
+        int index = indexOf(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            if (size == keys.length) {
+                K[] newKeys = (K[]) new Object[keys.length * 2];
+                V[] newValues = (V[]) new Object[values.length * 2];
+                System.arraycopy(keys, 0, newKeys, 0, keys.length);
+                System.arraycopy(values, 0, newValues, 0, values.length);
+                keys = newKeys;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:27]: nice, but resize is out of scope of the task anyway

**Review by okuzan — APPROVED** [review]

**Pull request closed without merging**